### ISSUE TITLE
.babelrc process.env.NODE_ENV targets

### ISSUE THREAD

**Issue by brigand** [babel/babel#1531]
Particularly for optimizations, it'd be good to configure babel per environment in .babelrc. If you want to do this currently you need to resort to hacks (`mv .babelrc.development .babelrc`), or not use .babelrc (or I'm missing something).

It could look like this:

```json
{
  "stage": 1,
  "env": {
    "production": {
      "loose": ["es6.modules", "es6.classes"],
    }
  }
}
```

And the behavior would essentially be `const {env, ...opts} = config; deepMerge(opts, env[process.env.NODE_ENV] || {})`.

**Comment by brigand** [comment]
My other thought is that there should be a BABEL_ENV which supersedes NODE_ENV, because some testing frameworks set NODE_ENV to 'test' always. And it should default to 'development' if neither is provided.

**Comment by sebmck** [comment]
Released as of 5.4.0. Follows your suggestion to the tee. Thanks!
